src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1760,13 +1768,15 @@ MultiNodeTree(Query *queryTree)
 	MultiProject *projectNode = NULL;
 	MultiExtendedOp *extendedOpNode = NULL;
 	MultiNode *currentTopNode = NULL;
+	DeferredErrorMessage *deferredError PG_USED_FOR_ASSERTS_ONLY = NULL;
 
 	/* verify we can perform distributed planning on this query */
 	ErrorIfQueryNotSupported(queryTree);
 
 	/* extract where clause qualifiers and verify we can plan for them */
 	whereClauseList = WhereClauseList(queryTree->jointree);
-	ValidateClauseList(whereClauseList);
+	deferredError = ValidateClauseList(whereClauseList);
+	Assert(deferredError == NULL);
```

Comment:
`MultiNodeTree` is called recursively. Are we sure we can never ever reach it when there is an unsupported clause type?